engine/api/src/main/java/io/deephaven/engine/table/ModifiedColumnSet.java

```diff
@@ -432,6 +432,15 @@ public int numColumns() {
         return columns.size();
     }
 
+    /**
+     * @return the names of the dirty columns in this set
+     */
+    public String[] dirtyColumnNames() {
+        if (dirtyColumns == null)
+            return new String[0];
```

Comment:
This could be `ArrayTypeUtils.EMPTY_STRING_ARRAY` to avoid creating a new empty array every time.